src/sagemaker/cli/compatibility/v2/modifiers/serde.py

```diff
@@ -0,0 +1,407 @@
+        return node
+
+
+class _ImportInserter(Modifier):
+    """A class to insert import statements into the Python module."""
+
+    def __init__(self, class_names, import_node):
+        """Initialize the ``class_names`` and ``import_node`` attributes.
+
+        Args:
+            class_names (set): If any of these class names are referenced in the
+                module, then ``import_node`` is inserted.
+            import_node (ast.ImportFrom): The AST node to insert.
+        """
+        self.class_names = class_names
+        self.import_node = import_node
+
+    def node_should_be_modified(self, module):
+        """Checks if the ``ast.Module`` node contains references to the
+        specified class names.
+
+        Args:
+            node (ast.Module): a node that represents a Python module. For more,
+                see https://docs.python.org/3/library/ast.html#abstract-grammar.
+
+        Returns:
+            bool: If the ``ast.Module`` references one of the specified classes.
+        """
+        for node in ast.walk(module):
+            if isinstance(node, ast.Name) and node.id in self.class_names:
+                return True
+            if isinstance(node, ast.Attribute) and node.attr in self.class_names:
+                return True
+        return False
+
+    def modify_node(self, module):
+        """Modifies the ``ast.Module`` node by inserted the specified node.
+
+        The ``import_node`` object is inserted immediately before the first
+        import statement.
+
+        Args:
+            node (ast.Module): a node that represents a Python module.
+
+        Returns:
+            ast.Module: a node that represents a Python module, which has been
+                modified to import a module.
+        """
+        for i, node in enumerate(module.body):
+            if isinstance(node, (ast.Import, ast.ImportFrom)):
+                module.body.insert(i, self.import_node)
+                return module
+
+        module.body.insert(0, self.import_node)
+        return module
+
+
+class SerializerImportInserter(_ImportInserter):
+    """A class to import the ``sagemaker.serializers`` module, if necessary.
+
+    This looks for references to the following classes:
+
+    - ``sagemaker.serializers.CSVSerializer``
+    - ``sagemaker.serializers.JSONSerializer``
+    - ``sagemaker.serializer.NumpySerializer``
+
+    Because ``SerializerImportInserter`` is gauranteed to run after
+    ``SerdeConstructorRenamer`` and ``SerdeObjectRenamer``,
+    we only need to check for the new serializer class names.
+    """
+
+    def __init__(self):
+        # Amazon SerDe are not defined in the sagemaker.serializers module.
+        class_names = {
+            class_name
+            for class_name in NEW_CLASS_NAMES - NEW_AMAZON_CLASS_NAMES
+            if "Serializer" in class_name
+        }
+        import_node = pasta.parse("from sagemaker import serializers\n").body[0]
```

Comment:
I think it'd be better to avoid `pasta` here as well

same below

Reply:
Fixed